src/components/state-lookup.ts

```diff
@@ -116,16 +116,21 @@ export class StateLookup {
     private async getInitialState(roomId: string): Promise<StateLookupRoom> {
         const r = this.dict[roomId];
         try {
-            const events: StateLookupEvent[] = await this.lookupQueue.add(
-                () => this._client.roomState(roomId)
+            const events = await this.lookupQueue.add(
+                () => this.intent.roomState(roomId, false) as Promise<StateLookupEvent[]>
             );
             events.forEach((ev) => {
                 this.insertEvent(r, ev);
             });
             return r;
         }
         catch (err) {
-            if (err.statusCode >= 400 && err.statusCode < 600) { // 4xx, 5xx
+            log.debug(`Failed to lookup state for room ${roomId}`, err);
+            const error = err as {statusCode?: number, message?: string};
+            if (error.message === "Failed to join room") {
+                throw err; // Could not join the room, don't retry.
+            }
+            if (error.statusCode && error.statusCode >= 400 && error.statusCode <= 599) { // 4xx, 5xx
```

Comment:
Unrelated to this PR, but shouldn't we retry `5xx`es?

Reply:
This PR plays it a bit safe, we could adapt it in the future.